esp-wifi/src/wifi/os_adapter.rs

```diff
@@ -868,6 +868,26 @@ pub unsafe extern "C" fn event_post(
     trace!("EVENT: {:?}", event);
     critical_section::with(|cs| WIFI_EVENTS.borrow_ref_mut(cs).insert(event));
 
+    // Call event handlers.
+    macro_rules! handle {
+        ($ty:ident) => {{
+            use crate::wifi::event::{$ty, WifiEventData};
+            assert_eq!(event_data_size, size_of::<$ty>(), "wrong size event data");
```

Comment:
This code belongs in `event.rs`, the macro shouldn't exist and this should be a method call on the event newtype. Or on `WifiEvent`.

Reply:
Which is better:
```rs
    use crate::wifi::event::*;
    // Safety: The `event_data` should be correct based on the event_id.
    match event {
        WifiEvent::ApProbereqrecved => {
            handle::<wifi_event_ap_probe_req_rx_t>(unsafe { &*event_data.cast() }, event_data_size)
        }
        WifiEvent::ApStaconnected => {
            handle::<wifi_event_ap_staconnected_t>(unsafe { &*event_data.cast() }, event_data_size)
        }
        WifiEvent::ApStadisconnected => handle::<wifi_event_ap_stadisconnected_t>(
            unsafe { &*event_data.cast() },
            event_data_size,
        ),
        _ => (),
    }
```
or
```rs
macro_rules! handle {
        ($ty:ident) => {
            crate::wifi::event::handle::<crate::wifi::event::$ty>(
                // Safety: The `event_data` should be correct based on the event_id.
                unsafe { &*event_data.cast() },
                event_data_size,
            )
        };
    }
    match event {
        WifiEvent::ApProbereqrecved => handle!(wifi_event_ap_probe_req_rx_t),
        WifiEvent::ApStaconnected => handle!(wifi_event_ap_staconnected_t),
        WifiEvent::ApStadisconnected => handle!(wifi_event_ap_stadisconnected_t),
        _ => (),
    }
```